rust/src/context.rs

```diff
@@ -78,6 +82,15 @@ impl InstrumentationContext {
             //     bail!("missed a function exit");
             // }
             func.end = SystemTime::now();
+            if func.end.duration_since(func.start)?.as_micros()
+                < self
```

Comment:
Another note: I think this is okay for now, but I feel like this filtering is more appropriate in the adapter phase. like, right before we turn the TraceEvent into the datadog formatted traces. We want as much of this computation and transformation to happen out of band. And it would be easier to do a lot of this with a complete tree.

Reply:
> nit: would read better just making a named variable for each of these right above here

updated naming for Go & JS well



> Another note: I think this is okay for now, but I feel like this filtering is more appropriate in the adapter phase. like, right before we turn the TraceEvent into the datadog formatted traces. We want as much of this computation and transformation to happen out of band. And it would be easier to do a lot of this with a complete tree.

I agree with that general direction especially as the filtering use cases become more involved.  We just need to keep in mind multi-tenant environments

Reply:
> We just need to keep in mind multi-tenant environments

should be okay. At the moment you need to make one adapter per tenant. We could fix this in the future by just linking the tenant specific config on the traceevent itself. like the adapter config and the filtering config.